draft-ietf-mls-protocol.md

```diff
@@ -1010,15 +1014,76 @@ A              B     ...      Z          Directory       Channel
+is also possible for a new member to join by means of an "external Commit".
+This mechanism can be used when the existing members don't have a KeyPackage for
+the new member, for example, in the case of an "open" group that can be joined
+by new members without asking permission from existing members.
+
+{{groupinfo-flow}} shows a typical  message flow for an external join. To enable
+a new member to join the group in this way, a member of the group (A, B)
+publishes a GroupInfo object that includes the GroupContext for the group as
+well as a public key that can be used to encrypt a secret to the existing
+members of the group.  When the new member Z wishes to join, they download the
+GroupInfo object and use it to form a Commit of a special form that adds Z to
+the group (as detailed in {{joining-via-external-commits}}).  The existing
+members of the group process this external Commit in a similar way to normal
+Commit, advancing to a new epoch in which Z is now a member of the group.
+
+~~~
+                                                          Group
+A              B              Z          Directory        Channel
+|              |              |              |              |
+| GroupInfo    |              |              |              |
++------------------------------------------->|              |
+|              |              | GroupInfo    |              |
+|              |              |<-------------+              |
+|              |              |              |              |
+|              |              | Commit(ExtZ) |              |
+|              |              +---------------------------->|
+|              |              |              | Commit(ExtZ) |
+|<----------------------------------------------------------+
+|              |<-------------------------------------------+
+|              |              |<----------------------------+
+|              |              |              |              |
+~~~
+{: #groupinfo-flow title"Client A publishes a GroupInfo object and Client Z uses
+it to join the group"}
+
+
 ## Relationships Between Epochs
 
 A group has a single linear sequence of epochs. Groups and epochs are generally
 independent of one another. However, it can sometimes be useful to link epochs
 cryptographically, either within a group or across groups. MLS derives a
 resumption pre-shared key (PSK) from each epoch to allow entropy extracted from
-one epoch to be injected into a future epoch. This link guarantees that members
-entering the new epoch agree on a key if and only if they were members of the group
-during the epoch from which the resumption key was extracted.
+one epoch to be injected into a future epoch.  A group member that wishes to
+inject a PSK issues a PreSharedKey proposal ({{presharedkey}}) describing the
+PSK to be injected.  When this proposal is committed, the corresponding PSK will
+be incorporated into the key schedule as described in {{pre-shared-keys}}.
+
+Linking epochs in this way
```

Comment:
strange line break

Reply:
Doesn't change the rendering.  I'll fix it if I end up having to do edits locally, vs in the GitHub UI :)